### PR TITLE
fix(enterprise): fix login flow

### PR DESCRIPTION
**Pull request by thsig**
**What this PR does / why we need it**:

Fixes a recent regression in the login flow, and stops throwing enterprise-related errors for non-enterprise projects when the user is logged in.

Also added unit tests for the `login` command.

### CODE REVIEW

File: garden-service/src/garden.ts [hunk 1]
```diff
@@ -329,25 +326,26 @@ export class Garden {
...
+    const enterpriseDomain = config.domain || null
     if (!opts.noEnterprise) {
       const enterpriseInitResult = await enterpriseInit({ log, projectConfig: config, environmentName })
       secrets = enterpriseInitResult.secrets
-      const clientAuthToken = enterpriseInitResult.clientAuthToken
-      if (clientAuthToken && projectId && enterpriseDomain) {
-        enterpriseContext = { clientAuthToken, projectId, enterpriseDomain }
-      }
+      clientAuthToken = enterpriseInitResult.clientAuthToken
+      // enterpriseContext = { clientAuthToken, projectId, enterpriseDomain: enterpriseDomain || null }
```
Comment (eysi09) [inline garden-service/src/garden.ts:337]: Remove this?

File: garden-service/test/unit/src/commands/login.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    footerLog: log,
+    args: {},
+    opts: withDefaultGlobalOpts({}),
+  }
+}
+
+describe("LoginCommand", () => {
+  let tmpDir: tmp.DirectoryResult
+  let projectConfig: ProjectConfig
+  // let sandbox = sinon.sandbox.create()
```
Comment (eysi09) [inline garden-service/test/unit/src/commands/login.ts:35]: Remove this.

File: garden-service/package.json [hunk 3]
```diff
@@ -215,6 +215,7 @@
     "prettier": "^1.19.1",
     "replace-in-file": "^6.0.0",
     "shx": "^0.3.2",
+    "sinon": "^9.0.3",
```
Comment (eysi09) [inline garden-service/package.json:218]: Do we need to install `sinon` for this? Why not just use `testdouble`?

Comment (thsig) [reply garden-service/package.json]: Good point—hadn't thought of that. These test cases now use `testdouble` (and I've gotten rid of `sinon`).

**Pull request merged**